lib/guardian/plug/ensure_not_authenticated.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Guardian.Plug.EnsureNotAuthenticated do
+  @moduledoc """
+  This plug ensures that a invalid JWT was provided and has been
+  verified on the request.
+
+  If one is not found, the `authenticated/2` function is invoked with the
+  `Plug.Conn.t` object and its params.
+
+  ## Example
+
+      # Will call the authenticated/2 function on your handler
+      plug Guardian.Plug.EnsureAuthenticated, handler: SomeModule
+
+      # look in the :secret location.  You can also do simple claim checks:
+      plug Guardian.Plug.EnsureAuthenticated, handler: SomeModule, key: :secret
+
+      plug Guardian.Plug.EnsureAuthenticated, handler: SomeModule, aud: "token"
+
+  If the handler option is not passed, `Guardian.Plug.ErrorHandler` will provide
+  the default behavior.
+  """
```

Comment:
What about if the callback was `already_authenticated/2`? Would that might make more sense as a callback in the error handler. Also, what would the default `Guardian.Plug.ErrorHandler` do when this callback was called?